### ISSUE TITLE
Zero Indexed Array Names

### ISSUE THREAD

**Issue by AdamBoxall** [danheberden/jquery-serializeForm#12]
Zero-indexed array names are not converted to an array. For example:

```
<input type="text" name="foo[0]" value="foo0">
<input type="text" name="foo[1]" value="foo1">
```

Is converted to:

```
"foo": {
    "0": "foo0",
    "1": "foo1"
}
```

The expected format would be:

```
"foo": [
    "foo0",
    "foo1"
]
```

**Comment by danheberden** [comment]
I'm confused why this would be the expected output? Using `[]` as a reference operator works for both strings and integers in JavaScript, and many languages will map `foo[0]` as either string or integer keys of that object/array. Can we get some common server-side conversions that happen to get an idea of what's typical?

**Comment by danheberden** [comment]
Sorry, didn't mean to close (damn close and comment button)

**Comment by danheberden** [comment]
Fixed fully with tests in `1.2.2` https://github.com/danheberden/jquery-serializeForm/commit/04b1cd3ca9958ed826e11751c3ac795b5bd6666a 

https://github.com/danheberden/jquery-serializeForm/commit/614618b0a36ebcac21632c9b3110c409282e85b3

https://github.com/danheberden/jquery-serializeForm/commit/c8c2695562c96dfe5a1e2e7348b874d97449f055
